starter-code/src/functions-and-arrays.js

```diff
@@ -1,100 +1,433 @@
 // Find the maximum
 
+function maxOfTwoNumbers(x, y) {
+  if (x > y) {
+    return x;
+  } else if (y > x) {
+    return y;
+  } else {
+    return x;
+  }
+}
+
 // Finding Longest Word
 var words = [
-  'mystery',
-  'brother',
-  'aviator',
-  'crocodile',
-  'pearl',
-  'orchard',
-  'crackpot'
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
 ];
 
+function findLongestWord(words) {
+  return words.sort(function(a, b) {
+    return b.length - a.length;
+  })[0];
+}
+console.log(findLongestWord(words));
+
+// function findLongestWord (words) {
+//   var longest = "";
+//   for (var i = 0; i < words.length; i += 1) {
+
+//   if (words[i].length > longest.length){
+//   longest = words[i];
+//   }
+
+//   }
+//   return longest
+//   }
+//   console.log(findLongestWord(words));
+
 // Calculating a Sum
 
 var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray(numbers) {
+  var sum = 0;
+  for (var i = 0; i < numbers.length; i += 1) {
+    sum += numbers[i];
+  }
+  return sum;
+}
+
+console.log(sumArray(numbers));
+
 // Calculate the Average
 
 var numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+function averageNumbers(numbersAvg) {
+  if (numbersAvg.length === 0) {
+    return undefined;
+  }
+  var resultAvg = sumArray(numbersAvg) / numbersAvg.length;
+  return resultAvg;
+}
 
 // Array of Strings
 var wordsArr = [
-  'seat',
-  'correspond',
-  'linen',
-  'motif',
-  'hole',
-  'smell',
-  'smart',
-  'chaos',
-  'fuel',
-  'palace'
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace"
 ];
 
+function averageWordLength(words) {
+  if (words.length === 0) {
+    return undefined;
+  }
+  var wordsString = words.toString();
+  var regex = /,/gi;
+  var wordsReplace = wordsString.replace(regex, "");
+  var avgWords = wordsReplace.length / words.length;
+  return avgWords;
+}
+
+console.log(averageWordLength(wordsArr));
+
```

Comment:
Ok, this is working. Here is another way to do it that I find easier to understand:
```
function averageWordLength(array) {
  if (array.length === 0) {
    return undefined;
  }
  var sumLetters = 0;
  for (var i = 0; i < array.length; i++) {
    sumLetters += array[i].length;
  }
  return sumLetters / array.length;
}
```